### PR TITLE
Azure doc

### PR DESCRIPTION
**Pull request by dylangrandmont**
Document Azure Blob Storage Integration steps

### CODE REVIEW

File: docs/advanced-usage/connecting-cloud-storage/azure-blob-storage.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+1. Click on the App registration you have created
+1. Note the "Tenant ID" and "Application (client) ID"
+1. Click "Certificates & secrets", then "New client secret"
+    1. Click "Add" to save this secret and note the key value
+
+### Step 2: Grant Permissions to App Reigration
+
+We will grant two permissions to the App registration created above:
+
+1. Storage Blob Delegator permission at the storage account level
```
Comment (sandalns) [inline docs/advanced-usage/connecting-cloud-storage/azure-blob-storage.md]: Should we link to the docs for the role? https://learn.microsoft.com/en-us/azure/role-based-access-control/built-in-roles#storage-blob-delegator

Comment (sandalns) [reply docs/advanced-usage/connecting-cloud-storage/azure-blob-storage.md]: https://learn.microsoft.com/en-us/azure/role-based-access-control/built-in-roles#storage-blob-data-reader

**Review by sandalns — APPROVED** [review]

File: docs/advanced-usage/connecting-cloud-storage/azure-blob-storage.md (hunk 1 above)
Comment (dylangrandmont) [reply docs/advanced-usage/connecting-cloud-storage/azure-blob-storage.md]: Good call 👍🏻

File: docs/advanced-usage/connecting-cloud-storage/azure-blob-storage.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    1. Click "Add" to save this secret and note the key value
+
+### Step 2: Assign Roles to App Registration
+
+We will assign two roles to the App registration created above:
+
+1. [Storage Blob Delegator](https://learn.microsoft.com/en-us/azure/role-based-access-control/built-in-roles#storage-blob-delegator) at the storage account level
+1. [Storage Blob Data Reader](https://learn.microsoft.com/en-us/azure/role-based-access-control/built-in-roles#storage-blob-data-reader) at the container level
+
+First we will assign the Storage Blob Delegator role:
```
Comment (gordonhart) [inline docs/advanced-usage/connecting-cloud-storage/azure-blob-storage.md]: Should each of these numbered lists live under their own subheader? Slightly confusing to have stepped lists 1-2, 1-6, and 1-6 all under "Step 2".

**Review by gordonhart — APPROVED** [review]

**Pull request merged**